Services/communication_service.c

```diff
@@ -17,7 +17,25 @@
  * @date 2020-09-25
  */
 
+#include "service_response.h"
+#include "service_utilities.h"
+#include "system.h"
 #include "communication_service.h"
+#include "comms_hal.h"
+#include "uhf_hal.h"
+
+static uint8_t SID_byte = 1;
+static Sband_config S_config;//In two different situations, globalization has resolved issues in set_freq
+Sband_Full_Status S_FS; // FS: Full Status
+
+UHF_Status U_stat;
+UHF_Beacon U_beacon;
+UHF_framStruct U_FRAM;
+UHF_Address U_I2C_add;
+UHF_Confirm U_restore;
+UHF_Confirm U_is_secure;
+UHF_Call_Sign U_callsign;
```

Comment:
So I tried them without being static and things worked. Is it better to make them static or not?
The problem that the local version makes is that when a packet arrives, it will go to `HL_sys_intvecs.asm` line 60 and you have to relaunch. It could happen when doing `cnv`, setting the packet length, or even after it exits the service.
The source might be a debugging issue .... I don't know.

Reply:
Yeah, static is preferred just to make it clear that they're not meant to be accessed elsewhere

Reply:
I think we should try again to make these local.
The issue is that these globals makes the code much more difficult to reason about in the multi-threaded environment and adds another layer where we have to be concerned with shared resource protection. For instance, if a command is sent to set the frequency and immediately after, a command is sent to query the frequency, this code might have trouble.

Reply:
Ok, I'll make them local and test them again and give you an update.

Reply:
Thanks, I just think it could save a world of pain later when we start to build the system up a bit more.